app/searchForm/index.js

```diff
@@ -0,0 +1,34 @@
+class searchFormController {
+    /*@ngInject*/
+    constructor(gitSearchService) {
+        this.query = '';
+        this.gitSearchService = gitSearchService;
+    }
+
+    getRequestParams() {
+        let parsedParams = {};
+        let params = this.query.split('+');
+        params.forEach((parameter) => {
+            parameter = parameter.trim();
+            let subParams = parameter.split(':');
+            if (subParams.length > 1) {
+                let trimmedParams = subParams.map(Function.prototype.call, String.prototype.trim);
```

Comment:
Second argument is an optional *this* argument ([check it here] (https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/map) )